[Qin Haichen] iP

QinHaichen12 opened the pull request:
# Bottle
> Roses are red, violets are blue, and my brain is blank

Bottle provides an avenue for you to put all your incomplete tasks - and procrastinate until they are long due.

It is
- Not very smart
- Not very sweet
- Not very stonks

However, I believe that it holds 2 redeeming qualities:
1. It remembers
2. It can count
3. What is number 1 again?

It will appear similar to something like this:
- [X] breathe
- [  ] be productive👍 

This is a *fun* [link](https://www.youtube.com/watch?v=dQw4w9WgXcQ) for you to check out!

Speaking of which, what does the `code` below do?
```
public class nani{
    static int huh(int n)
    {
        int F[][] = new int[][] { { 1, 1 }, { 1, 0 } };
        if (n == 0)
            return 0;
        power(F, n - 1);

        return F[0][0];
    }

    static void multiply(int F[][], int M[][])
    {
        int x = F[0][0] * M[0][0] + F[0][1] * M[1][0];
        int y = F[0][0] * M[0][1] + F[0][1] * M[1][1];
        int z = F[1][0] * M[0][0] + F[1][1] * M[1][0];
        int w = F[1][0] * M[0][1] + F[1][1] * M[1][1];

        F[0][0] = x;
        F[0][1] = y;
        F[1][0] = z;
        F[1][1] = w;
    }

    static void power(int F[][], int n)
    {
        if (n == 0 || n == 1)
            return;
        int M[][] = new int[][] { { 1, 1 }, { 1, 0 } };

        power(F, n / 2);
        multiply(F, F);

        if (n % 2 != 0)
            multiply(F, M);
    }

    public static void main(String args[])
    {
        int n = 9;
        System.out.println(huh(n));
    }
};
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
Add JUnit Test case

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

QinHaichen12 committed:
Level 0

QinHaichen12 committed:
Level 0

QinHaichen12 committed:
Level 1

QinHaichen12 committed:
Level 2

QinHaichen12 committed:
Level 3

QinHaichen12 committed:
Level 4

QinHaichen12 committed:
A-TextUiTesting

QinHaichen12 committed:
Level-5

QinHaichen12 committed:
Level-6

QinHaichen12 committed:
Level-7

QinHaichen12 committed:
Merge branch 'branch-Level-7'

QinHaichen12 committed:
Level-8

QinHaichen12 committed:
Merge branch 'branch-Level-8'

QinHaichen12 committed:
A-MoreOOP

QinHaichen12 committed:
Merge branch 'branch-A-MoreOOP'

QinHaichen12 committed:
Refactor the class

QinHaichen12 committed:
Merge branch 'branch-A-Packages'

QinHaichen12 committed:
Merge branch 'add-gradle-support'

QinHaichen12 committed:
Add gradle support

QinHaichen12 committed:
Add JUnit Test case

QinHaichen12 committed:
Merge branch 'A-JUnit'

QinHaichen12 committed:
Build v0.1

QinHaichen12 committed:
Add javadoc comments

QinHaichen12 committed:
Add javadoc comments

QinHaichen12 committed:
Add find function

QinHaichen12 committed:
Merge branch 'branch-A-JavaDoc'

QinHaichen12 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/bottle/Bottle.java
#	src/main/java/bottle/task/Task.java

Hoodini231 commented on src/main/java/bottle/Parser.java:
```diff
@@ -0,0 +1,127 @@
+            } else if (input.startsWith("deadline ")) {
+                String[] parts = input.split(" /by ");
+                if (parts.length != 2) {
+                    throw new IllegalArgumentException("OOPS!!! The deadline format is incorrect. Use: deadline <description> /by <dd/MM/yyyy HHmm>");
```
Line a bit long could use better indent formatting

Hoodini231 commented on src/main/java/bottle/command/invalidCommand.java:
```diff
@@ -0,0 +1,13 @@
+import bottle.exception.BottleException;
+import bottle.task.TaskList;
+
+public class invalidCommand extends Command {
```
wrong naming convention should be InvalidCommand
classes should be capitalized

Hoodini231 commented on src/test/java/bottle/task/TodoTest.java:
```diff
@@ -0,0 +1,59 @@
+
+    private Todo todo;
+
+    @BeforeEach
```
Could use comments / javadocs to more clearly explain what each test may do

Hoodini231 commented on src/main/java/bottle/Ui.java:
```diff
@@ -0,0 +1,112 @@
+     *
+     * @param taskList the task list
+     */
+    public void printTaskAddedMessage(ArrayList<Task> taskList) {
```
Diff naming convention than other print commands, this one has Message at the end but rest are print{context}
such as printMark vs printMarkMessage

Hoodini231 reviewed (commented):
Overall decent code and doesn't violet any major styling issues just do keep in mind class names need to be capitalized

DiegoTheExplorar commented on src/main/java/bottle/Bottle.java:
```diff
@@ -0,0 +1,36 @@
+
+import bottle.command.Command;
+import bottle.exception.BottleException;
+import bottle.task.*;
```
Hello there good sir. I believe we are supposed to avoid wildcard imports

DiegoTheExplorar commented on src/main/java/bottle/Bottle.java:
```diff
@@ -0,0 +1,36 @@
+import bottle.exception.BottleException;
+import bottle.task.*;
+
+import java.io.ByteArrayOutputStream;
```
Good job on separating these import statements.

DiegoTheExplorar commented on src/main/java/bottle/Bottle.java:
```diff
@@ -0,0 +1,36 @@
+
+import java.io.ByteArrayOutputStream;
+
+public class Bottle {
```
I saw that your other classes have javadoc but this doesn't. Do add it int :]

DiegoTheExplorar commented on src/main/java/bottle/Ui.java:
```diff
@@ -0,0 +1,112 @@
+     * The Welcome msg.
+     */
+    private final String welcomeMsg =
+            " Hello! I'm bottle.Bottle\n" +
+                    " What can I do for you?";
```
```suggestion
            " Hello! I'm bottle.Bottle\n" 
             + " What can I do for you?";
```

cause if not checkstyle screams at you :}

DiegoTheExplorar reviewed (commented):
Overall, noice job. Especially with the javadocs.

QinHaichen12 committed:
Format code

QinHaichen12 committed:
Merge branch 'branch-A-CheckStyle'

QinHaichen12 committed:
Add javafx files

QinHaichen12 committed:
Add GUI

Currently only CLI is present

GUI provides better User experience

Transferred Javafx files from sample tutorial and modified Command class to return string

String will be printed and displayed in GUI

QinHaichen12 committed:
Removed unused import statements

Some import statements are unnecessary

Removed them for better code quality

QinHaichen12 committed:
Merge branch 'branch-Level-10'

QinHaichen12 committed:
add assertions to methods

- Added assertions in constructors to verify that taskList is not null
- Added assertions in getTask and removeTask to check that the index is within bounds
- Added assertions in addTask to ensure tasks being added are not null
- Added assertions in toString to verify taskList is not null before iteration

These changes improve the robustness of the TaskList class by catching potential errors early during runtime.

QinHaichen12 committed:
Merge pull request #2 from QinHaichen12/branch-A-Assertions

add assertions to methods

QinHaichen12 committed:
Merge branch 'master' into branch-A-CodeQuality

QinHaichen12 committed:
improve Bottle class readability and maintainability

- Extracted command processing logic into a separate method `processCommand()` to reduce duplication in `run()` and `getResponse()`.
- Introduced `handleUserInput()` to encapsulate input handling logic in `run()`, improving clarity.
- Applied dependency injection for `Storage`, `Parser`, `TaskList`, and `Ui`, making the class more flexible and easier to test.
- Ensured the Bottle class adheres to the Single Level of Abstraction Principle (SLAP) by separating higher-level logic from lower-level details.
- Enhanced error handling by centralizing the command processing logic.

These changes improve the overall maintainability, readability, and testability of the `Bottle` class.

QinHaichen12 committed:
Merge pull request #3 from QinHaichen12/branch-A-CodeQuality

Branch a code quality

QinHaichen12 committed:
add detect anomalies extension to check for event scheduling conflicts

- Implemented the `detectAnomalies()` method to identify time overlaps between events in the task list.
- Extracted events from the task list and sorted them by start time to efficiently detect scheduling conflicts.
- Added logic to compare

QinHaichen12 committed:
Merge pull request #4 from QinHaichen12/branch-BCD-Extension

add detect anomalies extension to check for event scheduling conflicts

QinHaichen12 committed:
Improve GUI for bottle by adding css styles

Current GUI is very basic and lacks stying, making it difficult to read.

Adding colour and shape can enhance user's viewing experience. The FXML file is modified to include css styles.

This is good as it separates view from logic control.

QinHaichen12 committed:
Merge branch 'Branch-A-BetterGUI'

QinHaichen12 committed:
Merge branch 'master' of github.com:QinHaichen12/ip

QinHaichen12 committed:
Add additional error handling methods

Previous version was unable to print out error message when runtime exception occur

Displaying custom error messages can help the user understand the problem.

This is good as user is now able to understand why an invalid command is produced

QinHaichen12 committed:
Merge branch 'branch-A-MoreErrorHandling'

QinHaichen12 committed:
Add user guide

QinHaichen12 committed:
Change name of ui.png

Previously was ui.png and it is unable to be detected by the script

Changed the name to Ui.png

QinHaichen12 committed:
Change name of ui.png

Previously was ui.png and it is unable to be detected by the script

Changed the name to Ui.png

QinHaichen12 committed:
Fix style issues
